app/models/status_group.rb

```diff
@@ -0,0 +1,40 @@
+class StatusGroup
+  STATES_PRIORITY = %w(failure error pending success).freeze
+
+  attr_reader :statuses
+
+  def initialize(statuses)
+    @statuses = statuses
+  end
+
+  delegate :state, to: :main_status
+
+  def description
+    "#{success_count} / #{@statuses.count} checks OK"
+  end
+
+  def target_url
+  end
+
+  def main_status
+    @main_status ||= find_main_status
+  end
+
+  def to_partial_path
+    'statuses/group'
+  end
+
+  private
+
+  def find_main_status
+    STATES_PRIORITY.each do |state|
```

Comment:
Nevermind. While applying that change I remembered why it isn't a find. If it was this method would return a `String`, while we are looking for a `Status`.